src/PostgREST/Error.hs

```diff
@@ -93,9 +93,9 @@ instance JSON.ToJSON ApiRequestError where
     "message" .= T.decodeUtf8 errorMessage]
   toJSON InvalidRange = JSON.object [
     "message" .= ("HTTP Range error" :: Text)]
-  toJSON (NoRelBetween parent child) = JSON.object [
-    "hint"    .= ("If a new foreign key between these entities was created in the database, try reloading the schema cache." :: Text),
-    "message" .= ("Could not find a relationship between " <> parent <> " and " <> child <> " in the schema cache" :: Text)]
+  toJSON (NoRelBetween parent child schema) = JSON.object [
+    "hint"    .= ("Make sure that the tables '" <> parent <> "' and '" <> child <> "' exist in the database schema '" <> schema <> "' and that there is a foreign key relationship between them. If both conditions are met, try reloading the schema cache." :: Text),
```

Comment:
- Removing the tables mention, since the resources can be views as well
- Removing the database mention, which I think is unnecessary

```suggestion
    "hint"    .= ("Make sure that '" <> parent <> "' and '" <> child <> "' exist in the schema '" <> schema <> "' and that there is a foreign key relationship between them. If both conditions are met, try reloading the schema cache." :: Text),
```

Reply:
> If both conditions are met, try reloading the schema cache.

The above doesn't seem right, it tells the user to reload the schema cache after verifying.

Reply:
> [...] it tells the user to reload the schema cache after verifying.

Which seems like a perfectly valid procedure to me. Why do you think it's wrong?

Reply:
Ah, in the case the user detects a typo(`useRs` instead of `users`) then he/she would correct it and then proceed to reload the schema cache(following the hint verbatim) instead of just retrying the request.

Reply:
Also, I don't think where doing the hints right for some cases.

When `AmbiguousRelBetween` happens, we give a hint that's actionable from the frontend(change the url), while reloading the schema cache must be done from the backend - not actionable from the client side.

Reply:
> Ah, in the case the user detects a typo(useRs instead of users) then he/she would correct it and then proceed to reload the schema cache(following the hint verbatim) instead of just retrying the request.

... which would be fine, because it would be successful. I'd rather have somebody fix their queries **and** reload the schema cache, even if not needed, and be successful - then missing one of both an fail to get it right.

> When AmbiguousRelBetween happens, we give a hint that's actionable from the frontend(change the url), while reloading the schema cache must be done from the backend - not actionable from the client side.

I like that thought in principal. However, if we take the part about the schema cache out, then this might really leave people in the dark about what a possible solution would be.

Reply:
> Ah, in the case the user detects a typo(useRs instead of users) then he/she would correct it and then proceed to reload the schema cache(following the hint verbatim) instead of just retrying the request.

In that case,  we could keep the original hint at the end to avoid this.

```suggestion
    "hint"    .= ("Verify that '" <> parent <> "' and '" <> child <> "' exist in the schema '" <> schema <> "' and that there is a foreign key relationship between them. If a new relationship was created, try reloading the schema cache." :: Text),
```
Or maybe move the explanation to the `detail` key, like:

```suggestion
    "hint"    .= ("If a new foreign key between these entities was created in the database, try reloading the schema cache." :: Text),
    "detail"  .= ("There is no foreign key relationship between '" <> parent <> "' and '" <> child <> "' or they do not exist in the schema '"<> schema <> "'." :: Text),

> When AmbiguousRelBetween happens, we give a hint that's actionable from the frontend(change the url), while reloading the schema cache must be done from the backend - not actionable from the client side.

I think that's inevitable due to the hints that may come from database errors.

Reply:
> However, if we take the part about the schema cache out, then this might really leave people in the dark about what a possible solution would be.

True, I don't have a better suggestion ATM, so I think the schema cache should stay.

> In that case, we could keep the original hint at the end to avoid this.

@laurenceisla Yeah, giving the user multiple hints seems much better.